core/modules/widgets/widget.js

```diff
@@ -557,6 +557,29 @@ Widget.prototype.allowActionPropagation = function() {
 	return true;
 };
 
+Widget.prototype.generateTransclusionFootprint = function() {
+	var parentTransclusionWidget = this.findParentTransclusionWidget(),
+	    node = this,
+	    footprint = node.parentWidget.children.indexOf(node);
+	while(node = node.parentWidget) {
```

Comment:
I'm not keen on putting assignments within conditionals, as it's too easy to read it as a mistaken equality comparision. Is it intentional here?